### PR TITLE
Add new extension: Word Search

### PR DESCRIPTION
**Pull request by RishabSwift**
## Description

Word Search allows you to search for synonyms, antonyms, and rhymes given a word. It allows you to quickly look up words that fit your use case and can help your writing immensly.

## Type of change

- New extension

## Screencast

[Screencast Recording](https://cln.sh/rttkUH)

### Screenshots

Available on the [readme page](https://github.com/RishabSwift/extensions/tree/main/extensions/word-search): 

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/word-search/package.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  "title": "Word Search",
+  "description": "Search for synonyms, antonyms and rhymes",
+  "icon": "command-icon.png",
+  "author": "rishabswift",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "word-synonym-search",
+      "title": "Word Synonym Search",
+      "subtitle": "Search for synonyms",
```
Comment (pernielsentikaer) [inline extensions/word-search/package.json:13]:
```suggestion
      "subtitle": "Datamuse",
```

File: extensions/word-search/package.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      "description": "Search for synonyms given a word",
+      "mode": "view"
+    },
+    {
+      "name": "word-antonym-search",
+      "title": "Word Antonym Search",
+      "subtitle": "Search for antonyms",
```
Comment (pernielsentikaer) [inline extensions/word-search/package.json:20]:
```suggestion
      "subtitle": "Datamuse",
```

File: extensions/word-search/package.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      "description": "Search for antonyms given a word",
+      "mode": "view"
+    },
+    {
+      "name": "word-rhyme-search",
+      "title": "Word Rhyme search",
+      "subtitle": "Search for rhymes",
```
Comment (pernielsentikaer) [inline extensions/word-search/package.json:27]:
```suggestion
      "subtitle": "Datamuse",
```

File: extensions/word-search/src/search-results.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        <List.Item
+          icon={"command-icon.png"}
+          key={word.word}
+          title={word.word}
+          subtitle={word.defs !== undefined ? word.defs[0] : ""}
+          actions={
+            <ActionPanel>
+              <PasteAction
+                content={word.word}
+                title="Copy & Insert Word"
```
Comment (pernielsentikaer) [inline extensions/word-search/src/search-results.tsx:30]:
```suggestion
                title="Paste Word"
```
Also, you could also add [CopyToClipboardAction](https://developers.raycast.com/api-reference/user-interface/actions#copytoclipboardaction)

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @RishabSwift 👋 

Thanks for your contribution 💪 Thanks for the chat in Slack about your extension 💯 

I left some feedback for your PR that you can look into 🔥 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by RishabSwift — COMMENTED** [review]
Resolve suggested feedback

File: extensions/word-search/package.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    {
+      "name": "word-antonym-search",
+      "title": "Word Antonym Search",
+      "subtitle": "Datamuse",
+      "description": "Search for antonyms given a word",
+      "mode": "view"
+    },
+    {
+      "name": "word-rhyme-search",
+      "title": "Word Rhyme search",
```
Comment (pernielsentikaer) [inline extensions/word-search/package.json:26]:
```suggestion
      "title": "Search Word Rhyme",
```

File: extensions/word-search/package.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  "name": "word-search",
+  "title": "Word Search",
+  "description": "Search for synonyms, antonyms and rhymes",
+  "icon": "command-icon.png",
+  "author": "rishabswift",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "word-synonym-search",
+      "title": "Word Synonym Search",
```
Comment (pernielsentikaer) [inline extensions/word-search/package.json:12]:
```suggestion
      "title": "Search Word Synonym",
```

File: extensions/word-search/package.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      "subtitle": "Datamuse",
+      "description": "Search for synonyms given a word",
+      "mode": "view"
+    },
+    {
+      "name": "word-antonym-search",
+      "title": "Word Antonym Search",
```
Comment (pernielsentikaer) [inline extensions/word-search/package.json:19]:
```suggestion
      "title": "Search Word Antonym",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Just found a few more to correct 🔥

**Review by RishabSwift — COMMENTED** [review]
Updated suggested titles

**Review by pernielsentikaer — APPROVED** [review]
Hi @RishabSwift 👋 

Thanks for your contribution 🔥

**Comment by sxn** [comment]
hi @RishabSwift - nice extension! Just one small change and then we can merge it: could you move the screenshots out of the `assets` folder and into a `media/` folder in the root of the extension? The `assets/` folder is meant for files that need to be included with the extension, which for screenshots is not the case. 😸

**Comment by pernielsentikaer** [comment]
Are you sure? They are referenced in the README ☺

**Comment by pernielsentikaer** [comment]
> hi @RishabSwift - nice extension! Just one small change and then we can merge it: could you move the screenshots out of the `assets` folder and into a `media/` folder in the root of the extension? The `assets/` folder is meant for files that need to be included with the extension, which for screenshots is not the case. 😸 

Are you sure? They are referenced in the README ☺

**Comment by RishabSwift** [comment]
Hello @sxn 
Thanks for the feedbacak, I've moved the screenshots into a `media` folder and pushed it 😄

**Comment by sxn** [comment]
🎉

**Pull request merged**